domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientTest.java

```diff
@@ -709,5 +715,96 @@ public void getTotalRecordsTest() {
+    pc.getByIdAsStringForUpdate(mockConn, table, id, ar -> assertTrue(ar.succeeded()));
+    pc.getById(table, id, ar -> assertTrue(ar.succeeded()));
+    pc.getById(mockConn, table, id, ar -> assertTrue(ar.succeeded()));
+    pc.getByIdForUpdate(mockConn, table, id, ar -> assertTrue(ar.succeeded()));
+    pc.getById(table, id, Map.class, ar -> assertTrue(ar.succeeded()));
+    pc.getById(mockConn, table, id, Map.class, ar -> assertTrue(ar.succeeded()));
+    pc.getByIdForUpdate(mockConn, table, id, Map.class, ar -> assertTrue(ar.succeeded()));
+
+    // mock with query result
+    String jsonString = "{\"id\": \"abc\"}";
+    when(mockRowSet.size()).thenReturn(1);
+    @SuppressWarnings("unchecked")
+    RowIterator<Row> mockRowIterator = mock(RowIterator.class);
+    Row mockRow = mock(Row.class);
+    when(mockRowSet.iterator()).thenReturn(mockRowIterator);
+    when(mockRowIterator.next()).thenReturn(mockRow);
+    when(mockRow.getValue(anyInt())).thenReturn(jsonString);
+    // tests
+    pc.getByIdAsString(table, id, ar -> assertGetByIdAsString(ar));
+    pc.getByIdAsString(mockConn, table, id, ar -> assertGetByIdAsString(ar));
+    pc.getByIdAsStringForUpdate(mockConn, table, id, ar -> assertGetByIdAsString(ar));
+    pc.getById(table, id, ar -> assertGetByIdAsJson(ar));
+    pc.getById(mockConn, table, id, ar -> assertGetByIdAsJson(ar));
+    pc.getByIdForUpdate(mockConn, table, id, ar -> assertGetByIdAsJson(ar));
+    pc.getById(table, id, Map.class, ar -> assertGetByIdAsObject(ar));
+    pc.getById(mockConn, table, id, Map.class, ar -> assertGetByIdAsObject(ar));
+    pc.getByIdForUpdate(mockConn, table, id, Map.class, ar -> assertGetByIdAsObject(ar));
+    
+    // test exceptions
+    pc.getByIdAsString(Future.failedFuture("fail"), table, id, ar -> assertTrue(ar.failed()));
+    doAnswer(AdditionalAnswers.answerVoid((Handler<AsyncResult<PgConnection>> handler)
+        -> handler.handle(Future.failedFuture("fail"))))
+        .when(mockPgPool).getConnection(any());
+    pc.getByIdAsString(table, id, ar -> assertTrue(ar.failed()));
+    doAnswer(AdditionalAnswers.answerVoid(
+        (Tuple tuple, Handler<AsyncResult<RowSet<Row>>> handler)
+        -> handler.handle(Future.failedFuture("fail"))))
+        .when(mockPreparedQuery).execute(any(Tuple.class), any());
+    pc.getByIdAsString(mockConn, table, id, ar -> assertTrue(ar.failed()));
+    doAnswer(AdditionalAnswers.answerVoid(
+        (Tuple tuple, Handler<AsyncResult<RowSet<Row>>> handler)
+        -> handler.handle(Future.succeededFuture(mockRowSet))))
+        .when(mockPreparedQuery).execute(any(Tuple.class), any());
+    when(mockRow.getValue(anyInt())).thenThrow(new RuntimeException("fail"));
+    pc.getByIdAsString(mockConn, table, id, ar -> assertTrue(ar.failed()));
+    pc.getByIdAsString(mockConn, table, "1", ar -> assertTrue(ar.failed()));
+  } 
+  
+  private void assertGetByIdAsString(AsyncResult<String> ar) {
+    assertTrue(ar.succeeded());
```

Comment:
assertTrue(ar.succeeded())
should be handled by testContext.asyncAssertSuccess() or vertxTestContext.succeeding().
They also unwrap the result.